core/src/main/java/org/mule/runtime/core/privileged/processor/chain/AbstractMessageProcessorChain.java

```diff
@@ -227,6 +235,17 @@ public CoreEvent process(CoreEvent event) throws MuleException {
     }
   }
 
+  private void routeError(Consumer<Exception> errorRouter, Exception throwable) {
+    if (!isTransactionActive() && schedulerService.isCurrentThreadSwitchOnErrorRequired()) {
+      if (errorSwitchScheduler == null) {
```

Comment:
how could this be null? this should be done in initialization

Reply:
Moved to the `initialise` phase